pinot-core/src/main/java/org/apache/pinot/core/plan/maker/InstancePlanMakerImplV2.java

```diff
@@ -132,17 +140,40 @@ public InstancePlanMakerImplV2(QueryExecutorConfig queryExecutorConfig) {
     Preconditions
         .checkState(_groupByTrimThreshold > 0, "Invalid configurable: groupByTrimThreshold: %d must be positive",
             _groupByTrimThreshold);
-    LOGGER.info(
-        "Initializing plan maker with maxInitialResultHolderCapacity: {}, numGroupsLimit: {}, minSegmentGroupTrimSize: {}, minServerGroupTrimSize: {}",
-        _maxInitialResultHolderCapacity, _numGroupsLimit, _minSegmentGroupTrimSize, _minServerGroupTrimSize);
+    _enableBufferAcquireRelease = Boolean.parseBoolean(config.getProperty(ENABLE_ACQUIRE_AND_RELEASE));
+    LOGGER.info("Initializing plan maker with maxInitialResultHolderCapacity: {}, numGroupsLimit: {}, "
+            + "minSegmentGroupTrimSize: {}, minServerGroupTrimSize: {}, enableBufferAcquireRelease: {}",
+        _maxInitialResultHolderCapacity, _numGroupsLimit, _minSegmentGroupTrimSize, _minServerGroupTrimSize,
+        _enableBufferAcquireRelease);
   }
 
   @Override
   public Plan makeInstancePlan(List<IndexSegment> indexSegments, QueryContext queryContext,
       ExecutorService executorService, long endTimeMs) {
+    boolean prefetch = false;
+    if (_enableBufferAcquireRelease) {
+      prefetch = queryContext.getQueryOptions() != null && QueryOptions.isPrefetchBuffers(queryContext.getQueryOptions());
```

Comment:
Let's move the new logic into this `if` block, and keep the original logic in the `else` block to avoid always adding an extra operator
```suggestion
    List<PlanNode> planNodes = new ArrayList<>(indexSegments.size());
    if (_enableBufferAcquireRelease) {
      boolean prefetch = ...
    } else {
      for (IndexSegment indexSegment : indexSegments) {
        planNodes.add(makeSegmentPlanNode(indexSegment, queryContext));
      }
    }
```